baybe/insights/shap.py

```diff
@@ -0,0 +1,373 @@
+            """Explain first two data points to ensure that the explainer is working."""
+            if self._is_shap_explainer:
+                shap_explainer(self._bg_data.iloc[0:1])
+        except shap.utils._exceptions.InvalidModelError:
+            raise TypeError(
+                "The selected explainer class does not support the campaign surrogate."
+            )
+        except TypeError as e:
+            if (
+                "not supported for the input types" in str(e)
+                and not self._computational_representation
+            ):
+                raise NotImplementedError(
+                    "The selected explainer class does not support experimental "
+                    "representation. Switch to computational representation "
+                    "or use a different explainer "
+                    "(e.g. the default shap.KernelExplainer)."
+                )
+        return shap_explainer
+
+    def _init_explanation(
+        self,
+        data: pd.DataFrame | None = None,
+    ) -> shap.Explanation:
+        """Compute the Shapley values based on the chosen explainer and data set.
+
+        Args:
+            data: The data set for which the Shapley values should be computed.
+
+        Returns:
+            shap.Explanation: The computed Shapley values.
+
+        Raises:
+            ValueError: If the provided data set does not have the same amount of
+                parameters as the SHAP explainer background
+        """
+        if data is None:
+            data = self._bg_data
+        elif not self._bg_data.shape[1] == data.shape[1]:
+            raise ValueError(
+                "The provided data does not have the same amount of "
+                "parameters as the shap explainer background."
+            )
+
+        # Type checking for mypy
+        assert isinstance(data, pd.DataFrame)
+
+        if not self._is_shap_explainer:
+            """Return attributions for non-SHAP explainers."""
+            if self.explainer.__module__.endswith("maple"):
+                """Additional argument for maple to increase comparability to SHAP."""
+                attributions = self.explainer.attributions(
+                    data, multiply_by_input=True
+                )[0]
+            else:
+                attributions = self.explainer.attributions(data)[0]
+            explanations = shap.Explanation(
+                values=attributions,
+                base_values=self.explainer.model(self._bg_data).mean(),
+                data=data,
+                feature_names=data.columns.values,
+            )
+            return explanations
+        else:
+            explanations = self.explainer(data)
+
+        """Ensure that the explanation object is of the correct dimensionality."""
+        if len(explanations.shape) == 2:
+            return explanations
+        if len(explanations.shape) == 3:
+            return explanations[:, :, 0]

```

Comment:
Question: Whats going on here? Why needed? Can it be fixed by reshaping in case?